### PR TITLE
Remove star imports

### LINKED ISSUE #49416: A few years after #701 and PyTorch is still using implicit __all__ imports.
> i dont think we'll be fixing this. (i dont know if there's a way to do it either).

_Originally posted by @soumith in https://github.com/pytorch/pytorch/issues/701#issuecomment-329026903_

> > Maybe one could change `from torch._C import *` to `from torch._C import from_numpy, ...`?
> This seems simple enough. Are we certain this is the way to go? 
> I thought we'd need type hints (which would be way harder to convince people to use).
> Should we open another issue to discuss @moi90's solution?

_Originally posted by @cardoso-neto in https://github.com/pytorch/pytorch/issues/701#issuecomment-485272641_

A quick search for `import *` yields hundreds results.

If requested by a team member, I'd gladly do the tedious work replacing the import statements and pull requesting. :)
I would probably just need some pointers on how to do find out what is being imported from which C module, but I'd make quick work of it.
Long import statements could be indented very cleanly following [psf/black's](https://github.com/psf/black) style guides.

### PR DESCRIPTION
**Pull request by pmeier**
Fixes #49416.

Usage of star imports is determined by

```shell
$ git grep -n -E "import [*]" | grep -v __init__.py
```

### CODE REVIEW

File: test/test_determination.py [hunk 1]
```diff
@@ -62,11 +62,11 @@ def test_cpp_file(self):
     def test_test_file(self):
         """Test files trigger themselves and dependent tests"""
         self.assertEqual(
-            self.determined_tests(["test/test_jit.py"]), ["test_jit_profiling", "test_jit"]
+            self.determined_tests(["test/test_jit.py"]), ["test_jit"]
```
Comment (pmeier) [inline test/test_determination.py:65]: `test_jit_profiling` was only included to satisfy the test. After removing the star imports in `test/test_jit_profiling.py ` they are no longer picked up by this test and can be removed. cc @malfet

File: test/test_jit_fuser_legacy.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import sys
-sys.argv.append("--jit_executor=legacy")
-from test_jit_fuser import *  # noqa: F403
```
Comment (malfet) [inline test/test_jit_fuser_legacy.py:3]: Hmm, this would make this file a no op instead of rerunning all tests from test_jit_fuser

**Review by malfet — CHANGES_REQUESTED** [review]
Removing 'import *' from jit modes tests disables the testing.
Perhaps it can be reworked by importing main method from the appropriate tests...

File: test/test_jit_fuser_legacy.py (hunk 2 above)
Comment (pmeier) [reply test/test_jit_fuser_legacy.py:3]: Yeah, I was afraid of doing that. How exactly are these files run normally? I mean what command do I need to invoke so this gets run?

**Comment by InCogNiTo124** [comment]
What is the status of this?

**Comment by pmeier** [comment]
I can finish this after https://github.com/pytorch/pytorch/pull/61107#discussion_r662745341is resolved.

**Comment by d-miketa** [comment]
My Pylint brought me here today. It's unfortunate to see this PR unresolved after all this time and effort from the author. :/

**Pull request closed without merging**